src/main/java/core/basesyntax/figures/Figure.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.figures;
+
+public abstract class Figure implements Drawable {
+    protected String color;
+    protected double area;
+
+    protected Figure(String color) {
+        this.color = color;
+    }
+
+    public void draw() {
+        String[] fullClassName = this.getClass().getName().split("\\.");
+        String className = fullClassName[fullClassName.length - 1];
+        System.out.println("Figure: " + className);
+        System.out.println("Color: " + color);
+        System.out.println("Area: " + roundDouble(area) + " sq. units");
```

Comment:
Don`t use this way, it makes code less readable, every figure need have own implementation of draw()